### PR TITLE
fsck: support check or repair dir stat

### PR DESCRIPTION
**Pull request by Hexilee**

### CODE REVIEW

File: cmd/fsck.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ $ juicefs fsck redis://localhost --path /d1/d2 --recursive`,
 				Aliases: []string{"r"},
 				Usage:   "recursively check or repair",
 			},
+			&cli.BoolFlag{
+				Name:    "stat-all",
+				Aliases: []string{"s"},
```
Comment (davies) [inline cmd/fsck.go:69]: no short alias if it's not used frequently

Comment (Hexilee) [reply cmd/fsck.go]: Removed

File: pkg/meta/base.go [hunk 2]
```diff
@@ -295,7 +296,18 @@ func (m *baseMeta) calcDirStat(ctx Context, ino Ino) (space, inodes uint64, err
 }
 
 func (m *baseMeta) GetDirStat(ctx Context, inode Ino) (space, inodes uint64, err error) {
-	return m.en.doGetDirStat(ctx, m.checkRoot(inode))
+	var trySync bool
```
Comment (SandyXSD) [inline pkg/meta/base.go:299]: this variable is not necessary, use `!m.conf.ReadOnly` directly

File: pkg/meta/base.go [hunk 3]
```diff
@@ -1354,38 +1366,66 @@ func (m *baseMeta) Check(ctx Context, fpath string, repair bool, recursive bool)
 					// TODO
 					continue
 				}
+
+				var attrBroken, statBroken bool
 				if attr.Full {
 					nlink, st := m.countDirNlink(ctx, inode)
+					fmt.Printf("nlink of %s should be %d\n", path, nlink)
```
Comment (SandyXSD) [inline pkg/meta/base.go:1373]: remove it

File: pkg/meta/base.go [hunk 4]
```diff
@@ -1354,38 +1366,66 @@ func (m *baseMeta) Check(ctx Context, fpath string, repair bool, recursive bool)
...
-					logger.Warnf("nlink of %s should be %d, but got %d", path, nlink, attr.Nlink)
 				} else {
 					logger.Warnf("attribute of %s is missing", path)
+					attrBroken = true
+				}
+
+				stat, err := m.en.doGetDirStat(ctx, inode, false)
+				if err != nil {
+					logger.Errorf("get dir stat for inode %d: %v", inode, err)
+					continue
```
Comment (SandyXSD) [inline pkg/meta/base.go:1390]: attr may be broken and can be fixed.

File: cmd/fsck.go [hunk 5]
```diff
@@ -64,6 +64,10 @@ $ juicefs fsck redis://localhost --path /d1/d2 --recursive`,
 				Aliases: []string{"r"},
 				Usage:   "recursively check or repair",
 			},
+			&cli.BoolFlag{
+				Name:  "stat-all",
```
Comment (SandyXSD) [inline cmd/fsck.go:68]: need a more understandable flag name/usage.

File: pkg/meta/base.go (hunk 2 above)
Comment (Hexilee) [reply pkg/meta/base.go]: Removed

File: pkg/meta/base.go (hunk 3 above)
Comment (Hexilee) [reply pkg/meta/base.go]: Removed

File: pkg/meta/base.go (hunk 4 above)
Comment (Hexilee) [reply pkg/meta/base.go]: Fixed

File: cmd/fsck.go (hunk 5 above)
Comment (Hexilee) [reply cmd/fsck.go]: Renamed to `sync-dir-stat`

**Review by SandyXSD — APPROVED** [review]

**Pull request merged**